src/SessionSetup/ServiceDeployManager.cpp

```diff
@@ -560,9 +561,22 @@ void ServiceDeployManager::StartWatchdog() {
   ssh_watchdog_timer_.start(kSshWatchdogInterval);
 }
 
-outcome::result<ServiceDeployManager::GrpcPort> ServiceDeployManager::Exec() {
+outcome::result<ServiceDeployManager::GrpcPort> ServiceDeployManager::Exec(
+    orbit_metrics_uploader::MetricsUploader* metrics_uploader) {
+  orbit_metrics_uploader::ScopedMetric connect_metric{
+      metrics_uploader, orbit_metrics_uploader::OrbitLogEvent_LogEventType_ORBIT_INSTANCE_CONNECT};
+
   outcome::result<GrpcPort> result = outcome::success(GrpcPort{0});
   DeferToBackgroundThreadAndWait(this, [&]() { result = ExecImpl(); });
+
+  if (result.error() == make_error_code(Error::kUserCanceledServiceDeployment)) {
+    connect_metric.SetStatusCode(orbit_metrics_uploader::OrbitLogEvent_StatusCode_CANCELLED);
+    LOG("OrbitService deployment has been aborted by the user");
+  } else {
+    connect_metric.SetStatusCode(orbit_metrics_uploader::OrbitLogEvent_StatusCode_INTERNAL_ERROR);
+    LOG("Deployment successful, grpc_port: %d", result.value().grpc_port);
+  }
```

Comment:
Here is something wrong. There are three possible outcomes of `ExecImpl` 
1. Success, in which case the `connect_metric` should either not be touched at all (the default status code is 
success), or should be set to SUCCESS explicitly.
2. The result carries an error, but the error code is kUserCanceledServiceDeployment. In this case : CANCELLED
3. The result is a different error, then its INTERNAL_ERROR

Also. Before `if (result.error())` there needs to be a `if (result.has_error())` or `if(!result)`

Reply:
Thanks for catching this, I completely missed that!